receiver/k8sobjectsreceiver/config.go

```diff
@@ -50,53 +58,26 @@ type K8sObjectsConfig struct {
 type Config struct {
 	k8sconfig.APIConfig `mapstructure:",squash"`
 
-	Objects []*K8sObjectsConfig `mapstructure:"objects"`
+	Objects   []*K8sObjectsConfig `mapstructure:"objects"`
+	ErrorMode ErrorMode           `mapstructure:"error_mode"`
+	logger    *zap.Logger
 
 	// For mocking purposes only.
 	makeDiscoveryClient func() (discovery.ServerResourcesInterface, error)
 	makeDynamicClient   func() (dynamic.Interface, error)
 }
 
-func (c *Config) Validate() error {
```

Comment:
I suggest this refactoring to take place in a separate PR.

Reply:
Makes sense. I can revert to [c5d68b5](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38851/commits/c5d68b55356c601de22bbd6935d9a3fb19bc899e) move the default propagation into default config, and if we are fine with ignore/silent/propagate (can also reduce to true/false or similar) the PR should fix the original validation issue at startup.